src/ExtCmd.actor.cpp

```diff
@@ -404,7 +404,77 @@ struct DropCollectionCmd {
 		return doDropCollection(ec, query, reply);
 	}
 };
-REGISTER_CMD(DropCollectionCmd, "drop");
+REGISTER_CMD(DropCollectionCmd, "drop"); 
+ 
+ACTOR static Future<int> internal_doRenameCollectionIndexesActor(Reference<DocTransaction> tr, 
+                                                     Namespace ns, 
+                                                     Reference<MetadataManager> mm) { 
+    state Reference<UnboundCollectionContext> unbound = wait(mm->getUnboundCollectionContext(tr, ns)); 
+    unbound->bindCollectionContext(tr)->bumpMetadataVersion(); 
+    TraceEvent(SevInfo, "BumpMetadataVersion") 
+        .detail("reason", "renameCollection") 
+        .detail("ns", fullCollNameToString(ns)); 
+    return true; 
+} 
+ 
+ACTOR static Future<Void> Internal_doRenameCollection(Reference<DocTransaction> tr, 
+                                                    Reference<ExtMsgQuery> query, 
+                                                    Reference<ExtConnection> ec) { 
+    state Namespace ns; 
+    ns.first = upOneLevel(query->ns.second); 
+    state std::string sourceCollection = getLastPart(query->ns.second); 
+    state std::string destinationCollection = getLastPart(query->query.getStringField("to")); 
+    state bool dropTarget = query->query.getBoolField("dropTarget"); 
+    state bool exists_destinationCollection = wait(ec->docLayer->rootDirectory->exists(tr->tr, 
+                            {StringRef(ns.first), StringRef(destinationCollection)})); 
+    state bool exists_sourceCollection = wait(ec->docLayer->rootDirectory->exists(tr->tr, 
+                            {StringRef(ns.first), StringRef(sourceCollection)})); 
+ 
+    if(exists_sourceCollection) { 
```

Comment:
Also, probably we can short circuit here and return error if `exists_sourceCollection == false` ?

Reply:
ok, will check it.